ci-operator/step-registry/vsphere/provision/dnsmasq/vsphere-provision-dnsmasq-commands.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+SSH_PRIV_KEY_PATH="${CLUSTER_PROFILE_DIR}/ssh-privatekey"
+BASTION_IP=$(<"${SHARED_DIR}/bastion_private_address")
+
+cat > "${SHARED_DIR}/ansible-bastion-host" << EOF
+[all:vars]
+ansible_ssh_common_args="-o IdentityFile=${SSH_PRIV_KEY_PATH} -o StrictHostKeyChecking=no"
+ansible_user=core
+ansible_become=True
+
+[bastion]
+${BASTION_IP}
+EOF
+
+cat > start-dnsmasq-service.yaml <<-'EOF'
+- name: ensure dnsmasq service is started
+  hosts: bastion
+  any_errors_fatal: true
+  gather_facts: false
+  tasks:
+  - name: start dnsmasq service
```

Comment:
the bastion is a dynamic-provisioned rhcos vm, right?

Reply:
@jinyunma  the dnsmq is already installed by default in the rhcos image, just not started yet?

Reply:
yes, exact, installed but not started.

Reply:
Nice, save us a big effort.